webpack/assets/javascripts/react_app/components/PF4/InlineEdit/InlineEdit.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Spinner,
+  TextInput,
+  TextArea,
+  Text,
+  TextVariants,
+  Button,
+  Split,
+  SplitItem,
+} from '@patternfly/react-core';
+import { TimesIcon, CheckIcon, PencilAltIcon } from '@patternfly/react-icons';
+import PropTypes from 'prop-types';
+import { translate as __ } from '../../../common/I18n';
+import './inlineEdit.scss';
+
+const InlineEdit = ({ onEdit, value, textArea, attribute }) => {
+  // Tracks input box state
+  const [inputValue, setInputValue] = useState(value);
+  const [editing, setEditing] = useState(false);
+  const [submitting, setSubmitting] = useState(false);
+
+  // Setting didCancel to avoid actions from happening after component has been unmounted
+  // see https://overreacted.io/a-complete-guide-to-useeffect/#speaking-of-race-conditions
+  useEffect(() => {
+    let didCancel = false;
+
+    const onSubmit = async () => {
+      if (submitting) {
+        await onEdit(inputValue, attribute);
+        if (!didCancel) {
+          setSubmitting(false);
+          setEditing(false);
+        }
+      }
+    };
+    onSubmit();
+
+    return () => {
+      didCancel = true;
+    };
+  }, [submitting]);
+
+  // Listen for enter and trigger submit workflow on enter
+  useEffect(() => {
+    const listener = event => {
+      if (event.code === 'Enter' || event.code === 'NumpadEnter') {
+        if (editing) setSubmitting(true);
```

Comment:
@ezr-ondrej I agree, if it's a `<TextArea>` I think I should be able to use the Enter key and type multi-line text.  Could add a check for it here.

Reply:
I'd still want to know why we want to go in such a trouble to have the `Enter` key working, when it's not such a struggle to submit the inline form without it.

Reply:
@ezr-ondrej I think it does make sense to have it working, since it provides feature parity with existing inputs all over Foreman and Katello.  Examples:  Foreman Settings page - you can press enter to submit after changing a value.  Katello Product details: you can press enter to submit when editing the name, but you can use enter for multiline text in the description.

Reply:
The real question is why doesn't Patternfly provide this out of the box...

Reply:
I think submitting on Enter is a pretty common use case on web applications and something we should standardize in our application, we are trying to improve our UX and have historically not been so ergonomic to use :)